File: 剑指Offer/5.4跳跃游戏.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        maxPos = 0
        numsLen = len(nums)
        end = 0
        step = 0

        for i in range(numsLen - 1):
            if maxPos >= i:
                maxPos = max(maxPos, nums[i] + i);
                if i == end:
                    end = maxPos
                    step += 1

        return step

File: 剑指Offer/5.4跳跃游戏.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int maxPos = 0, numsLen = nums.size(), end = 0, step = 0;
        for (int i = 0; i < numsLen - 1; i++) {
            if (maxPos >= i) {
                maxPos = max(maxPos, i + nums[i]);
                if (i == end) {
                    end = maxPos;
                    step++;
                }
            }
        }
        return step;
    }
};